modules/nextflow/src/test/groovy/nextflow/splitter/CsvSplitterTest.groovy

```diff
@@ -150,6 +159,42 @@ class CsvSplitterTest extends Specification {
         items[2].z == 'iota'
     }
 
+    def testSplitCsvWithTypesAuto() {
+
+        when:
+        def items = new CsvSplitter().target(typesText).options(header:true,types:true).list()
+        then:
+        items.size() == 4
+
+        items[0].string == 'gamma' && items[0].string instanceof String
+        items[0].integer == 1 && items[0].integer instanceof Integer
+        items[0].boolean == true && items[0].boolean instanceof Boolean
+        items[0].float == 1.1.toDouble() && items[0].float instanceof Double
```

Comment:
Not sure, there's a need to implement this. Groovy provides out of the box helper to convert to any type using `toInteger()`, `toLong()`, `toBoolean` etc

Reply:
Not sure what you mean? The PR is using those methods to do the type casting.

Reply:
why not using buildit `toString()`, `toInteger()`, etc

Reply:
I understand that, but I do not understand the above snippet does

Reply:
Hello :) I am using those to convert the types [here](https://github.com/nextflow-io/nextflow/pull/3831/files/8e076089ce2d8a01cfe075ae33f39da620fca254#diff-32144a1a47a0ea418938e70e52a8fe650515c61d94b3d31fc421067434430ac5R234). Is that what you are referring to?